src/hazardous/hash/blake2/blake2b.rs

```diff
@@ -62,17 +62,29 @@
 //! [`mac::blake2b`]: crate::hazardous::mac::blake2b
 
 use crate::errors::UnknownCryptoError;
+use crate::hazardous::base::{Public, PublicArray, Wrapper};
 use crate::hazardous::hash::blake2::blake2b_core;
 use crate::hazardous::hash::blake2::blake2b_core::BLAKE2B_OUTSIZE;
 
-construct_public! {
-    /// A type to represent the `Digest` that BLAKE2b returns.
-    ///
-    /// # Errors:
-    /// An error will be returned if:
-    /// - `slice` is empty.
-    /// - `slice` is greater than 64 bytes.
-    (Digest, test_digest, 1, BLAKE2B_OUTSIZE)
+type Blake2bArray = PublicArray<1, BLAKE2B_OUTSIZE>;
+
+/// A type to represent the `Digest` that BLAKE2b returns.
+///
+/// # Errors:
+/// An error will be returned if:
+/// - `slice` is empty.
+/// - `slice` is greater than 64 bytes.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Digest(Blake2bArray);
+
+impl Wrapper<Blake2bArray> for Digest {
+    fn data(&self) -> &Blake2bArray {
+        &self.0
+    }
+
+    fn from(data: Blake2bArray) -> Self {
+        Self(data)
+    }
```

Comment:
If we choose not to use macros anywhere, we will incur this cost of extra boilerplate code for each newtype we want to add. It's not that much boilerplate, but it may add up. The simple solution here is to have a single small macro (`impl_wrapper`) that just does the `Wrapper<T>` implementation, and let the traits take care of the rest.